### ISSUE TITLE
MAINT: iid=True in `cov.py` is deprecated

### ISSUE THREAD

**Issue by larsoner** [mne-tools/mne-python#6312]
On latest `sklearn` master I get this in tests:
```
mne/cov.py:991: in _compute_covariance_auto
    gs.fit(data_[:, picks])
...
E           DeprecationWarning: The parameter 'iid' is deprecated in 0.22 and will be removed in 0.24.
```
Which is really from `cov.py` -- here showing where `gs` is instantiated and used:
```
            elif method_ == 'shrunk':
                ...
                gs = GridSearchCV(ShrunkCovariance(**mp),
                                  tuned_parameters, cv=cv, iid=True)
                for ch_type, picks in sub_picks_list:
                    gs.fit(data_[:, picks])
                    shrinkages.append((ch_type, gs.best_estimator_.shrinkage,
                                       picks))
```
I was hoping this would be an easy `**kwargs`-based-on-version fix, but it looks like `iid` is going away -- from the [sklearn docs](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html):
> iid : boolean, default=’warn’
>    If True, return the average score across folds, weighted by the number of samples in each test set. In this case, the data is assumed to be identically distributed across the folds, and the loss minimized is the total loss per sample, and not the mean loss across the folds. If False, return the average score across folds. Default is True, but will change to False in version 0.22, to correspond to the standard definition of cross-validation.
>    Changed in version 0.20: Parameter iid will change from True to False by default in version 0.22, and will be removed in 0.24.

@agramfort what do we do here? Is `iid=True` important, or is setting it to `iid=False` okay?

**Comment by agramfort** [comment]
we should switch to iid=False

>
